internal/runtime/client/client.go

```diff
@@ -250,12 +365,22 @@ func httpCall(ctx context.Context, request, response runtime.Object, opts *httpC
 	}
 	resp, err := client.Do(httpRequest)
 	if err != nil {
-		return errors.Wrap(err, "failed to perform the http call")
+		return errCallingExtensionHandler(
+			errors.Wrapf(err, "failed to call ExtensionHandler: %q", opts.name),
+		)
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return errCallingExtensionHandler(
+			fmt.Errorf("non 200 response code, %q, not accepted", resp.StatusCode),
```

Comment:
```suggestion
			errors.Errorf("non 200 response code, %q, not accepted", resp.StatusCode),
```
nit (for consistent library usage)